client/state/ui/layout-focus/reducer.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * External dependencies
+ */
+import includes from 'lodash/includes';
+
+/**
+ * Internal dependencies
+ */
+import {
+	LAYOUT_SET_FOCUS,
+	LAYOUT_NEXT_FOCUS_ACTIVATE,
+	LAYOUT_SET_NEXT_FOCUS,
+	SERIALIZE,
+	DESERIALIZE,
+} from 'state/action-types';
+import layoutFocusModule from 'lib/layout-focus';
+
+const initialState = { current: null, previous: null, next: null };
+
+export default function layoutFocus( state = initialState, action ) {
+	const validAreas = [ 'content', 'sidebar', 'sites', 'preview' ];
```

Comment:
Should we move this to a top-of-module constant so we're not redeclaring it every time the reducer is called?